src/TitleDetails/BoundPiecesList/hooks/useItemsList/useItemsList.js

```diff
@@ -0,0 +1,55 @@
+import { useQuery } from 'react-query';
+
+import {
+  batchRequest,
+  ITEMS_API,
+  LIMIT_MAX,
+  ORDER_PIECES_API,
+} from '@folio/stripes-acq-components';
+import {
+  useNamespace,
+  useOkapiKy,
+} from '@folio/stripes/core';
+
+const DEFAULT_DATA = [];
+
+export const useItemsList = ({ titleId, poLineId, options = {} }) => {
+  const { enabled = true, ...otherOptions } = options;
+  const ky = useOkapiKy();
+  const [namespace] = useNamespace({ key: 'bound-items-list' });
+  const boundItemsQuery = `titleId==${titleId} and poLineId==${poLineId} and isBound==true sortby receivedDate`;
+
+  const searchParams = {
+    limit: LIMIT_MAX,
+    query: boundItemsQuery,
+  };
+
+  const {
+    data = DEFAULT_DATA,
+    isLoading,
+    isFetching,
+    refetch,
+  } = useQuery({
+    queryKey: [namespace, titleId, poLineId],
+    queryFn: async () => {
+      const { pieces = DEFAULT_DATA } = await ky.get(ORDER_PIECES_API, { searchParams }).json();
```

Comment:
I believe using pagination for batch fetching may not be the best choice because you can't guarantee the exact number of items returned by each request. Additionally, sorting in the query doesn't seem to serve a meaningful purpose in this context.

Reply:
Updated and upload a screen record